monitoring-dashboard/components/org.wso2.ei.dashboard.bootstrap/src/main/java/org/wso2/ei/dashboard/bootstrap/DashboardServer.java

```diff
@@ -148,24 +145,58 @@ public void startServerWithConfigs() {
             System.exit(1);
         } catch (ConfigParserException e) {
             logger.error("Error while reading TOML file configs", e);
+        } catch (IllegalArgumentException e) {
+            logger.error(e.getMessage(), e);
+            System.exit(1);
         }
 
         Server server = new Server();
-        setServerConnectors(serverPort, server, DASHBOARD_HOME);
-        setServerHandlers(DASHBOARD_HOME, server);
-        addShutdownHook();
-
+        configureServer(server, serverPort, serverProtocol);
         try {
-            server.start();
-            writePID(DASHBOARD_HOME);
-            printServerStartupLog(serverPort);
-            server.join();
+            startAndMonitorServer(server, serverPort, serverProtocol);
         } catch (Exception ex) {
             logger.error("Error while starting up the server", ex);
         }
         logger.info("Stopping the server");
     }
 
+    private int getServerPort(Map<String, Object> parsedConfigs, int defaultPort) {
+        Object serverPortConfig = parsedConfigs.get(TOML_CONF_PORT);
+        if (serverPortConfig instanceof Long) {
+            return ((Long) serverPortConfig).intValue();
+        }
+        return defaultPort;
+    }
+
+    private String getServerProtocol(Map<String, Object> parsedConfigs) {
+        Object serverProtocolConfig = parsedConfigs.get(TOML_CONF_PROTOCOL);
+        if (serverProtocolConfig instanceof String) {
+            String protocol = (String) serverProtocolConfig;
+            if (HTTPS_PROTOCOL.equalsIgnoreCase(protocol)) {
+                return HTTPS_PROTOCOL;
+            }
+            if (HTTP_PROTOCOL.equalsIgnoreCase(protocol)) {
+                return HTTP_PROTOCOL;
+            }
+            throw new IllegalArgumentException("Invalid value provided for "
+                    + TOML_CONF_PROTOCOL + ": expected either \"http\" or \"https\"");
+        }
```

Comment:
Does this give an error if the protocol is not a string, let's say a number?

Reply:
No it won't it will use the HTTPS as default protocol in that case. Will update the code to throw an exception